pyam/utils.py

```diff
@@ -309,10 +309,15 @@ def pattern_match(data, values, level=None, regexp=False, has_nan=True):
     matching of model/scenario names, variables, regions, and meta columns to
     pseudo-regex (if `regexp == False`) for filtering (str, int, bool)
     """
-    matches = np.array([False] * len(data))
+    # coerce values to a list
     if not isinstance(values, collections.Iterable) or isstr(values):
         values = [values]
 
+    if len(data) == 0 or len(values) == 0:
```

Comment:
@gidden you return here a "normal" python array - would you expect it can provide same contacts are np.ndarray?